test/Microsoft.AspNetCore.Mvc.RazorPages.Test/Infrastructure/ServiceBasedPageModelActivatorProviderTest.cs

```diff
@@ -0,0 +1,151 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Reflection;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Testing;
+using Moq;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Mvc.RazorPages.Infrastructure
+{
+    public class ServiceBasedPageModelActivatorProviderTest
+    {
+        [Fact]
+        public void CreateActivator_ThrowsIfModelTypeInfoOnActionDescriptorIsNull()
+        {
+            // Arrange
+            var activatorProvider = new ServiceBasedPageModelActivatorProvider();
+            var descriptor = new CompiledPageActionDescriptor();
+
+            // Act & Assert
+            ExceptionAssert.ThrowsArgument(
+                () => activatorProvider.CreateActivator(descriptor),
+                "descriptor",
+                "The 'ModelTypeInfo' property of 'descriptor' must not be null.");
+        }
+
+        [Fact]
+        public void Create_GetsServicesFromServiceProvider()
+        {
+            // Arrange
+            var simpleModel = new DISimpleModel();
+            var serviceProvider = new Mock<IServiceProvider>(MockBehavior.Strict);
+            serviceProvider.Setup(s => s.GetService(typeof(DISimpleModel)))
+                           .Returns(simpleModel)
```

Comment:
Formatting: Could you untab this so it's where the IDE puts them by default? e.g.

```
serviceProvider.Setup(s => s.GetService(typeof(DISimpleModel)))
    .Returns(simpleModel)
    .Verifiable();
```